design/ceph/object/store.md

```diff
@@ -176,3 +183,50 @@ spec:
   gateway:
     caBundleRef: #ldaps-cabundle
 ```
+
+### Virtual Host Style access for buckets
+
+The Ceph Object Gateway supports accessing buckets using [virtual host style](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html) which allows accessing buckets using the bucket name as a subdomain in the endpoint. The user can configure the option via `radosgw-admin` like below:
+
+```sh
+# ceph config set client.rgw.mystore.a rgw_dns_name http://my-ingress.mydomain.com,rook-ceph-rgw-my-store.rook-ceph.svc
+```
+
+The user need to restart the RGW pods for the changes to take effect. Also multiple endpoints can be added to the list separated by comma. This is supported from Ceph Reef release(v18.0) onwards. This is different from `customEndpoints` which is used for configuring the object store to replicate and sync data amongst Multisite. For accessing the bucket point user need to configure wildcard dns in the cluster using [ingress loadbalancer](https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards) or in openshift cluster use [dns operator](https://docs.openshift.com/container-platform/4.9/networking/dns-operator.html). Same for TLS certificate, user need to configure the TLS certificate for the wildcard dns for the RGW endpoint. This option won't be enabled by default, user need to enable it by adding the `hosting` section in the `Gateway` settings:
```

Comment:
I have already suggested that Rook should apply the `--rgw-dns-name="comma,separated,list"` to the RGW deployment so that they are restarted automatically, requiring no user action.